packages/api-db-mongodb/src/db/comment.ts

```diff
@@ -210,14 +213,24 @@ export class MongoDBCommentAdapter implements DBCommentAdapter {
         .match({
           $and: [{itemID: id}, {state: CommentState.Approved}]
         })
+        .toArray(),
+      this.comments
+        .aggregate([{$sort: {'modifiedAt.date': -1}}], {
+          collation: {locale: this.locale, strength: 2}
+        })
+        .match({
+          $and: [{itemID: id}, {userID}, {state: {$ne: CommentState.Approved}}]
```

Comment:
i think you need to check if `userID` is undefined. Otherwise you'll get all here all comments that are not approved.

Reply:
I guess if it has no value "undefined" it won't match with any comments, and I tested it and seems working :S But what do you think?

Reply:
This is right but we don't know how this will evolve maybe in the future we have anonymous comments with userID undefined. Also performance wise it's faster to not execute the query if we don't have a userID.

Reply:
totally agree